ts/src/program/namespace/account.ts

```diff
@@ -184,6 +205,45 @@ export class AccountClient<
     });
   }
 
+  /**
+   * Returns multiple deserialized program derived accounts and their PublicKeys.
+   * Accounts not found or with wrong discriminator are returned as null.
+   *
+   * @param seedsArray The array of seeds of the accounts to fetch.
+   */
+  async fetchMultiplePda(
+    seedsArray: Array<Buffer | Uint8Array>[],
+    commitment?: Commitment
+  ): Promise<[(Object | null)[], PublicKey[]]> {
+    const addresses = (
+      await Promise.all(
+        seedsArray.map((seeds) =>
+          PublicKey.findProgramAddress(seeds, this._programId)
+        )
+      )
+    ).map(([address]) => address);
+
+    const accounts = await rpcUtil.getMultipleAccounts(
+      this._provider.connection,
+      addresses.map(translateAddress),
+      commitment
+    );
+
+    // Decode accounts where discriminator is correct, null otherwise
+    return [
+      accounts.map((account) => {
+        if (account == null) {
+          return null;
+        }
+        return this._coder.accounts.decode(
+          this._idlAccount.name,
+          account?.account.data
+        );
+      }),
+      addresses,
```

Comment:
Instead of returning a two element array, we should return an object here to be consistent with `rpcUtil.getMultipleAccounts`.

Reply:
Ok, I see `publicKey` was already the key in the Object.